src/components/policy/policy_regular/src/policy_manager_impl.cc

```diff
@@ -173,17 +182,121 @@ void FilterInvalidParameters(policy_table::RpcParameters& rpc_parameters) {
   rpc_parameters.parameters->swap(valid_params);
 }
 
+/**
+ * @brief FilterInvalidNotifications filter notification priorities that are not
+ * present in schema
+ * @param notification priorities to filter
+ */
+void FilterInvalidNotifications(
+    policy_table::NumberOfNotificationsPerMinute& notifications) {
+  policy_table::NumberOfNotificationsPerMinute valid_notifications;
+  for (const auto& notification : notifications) {
+    policy_table::Priority priority;
+    if (policy_table::EnumFromJsonString(notification.first, &priority)) {
+      valid_notifications.insert(notification);
+    }
+  }
+  notifications.swap(valid_notifications);
+}
+
+/**
+ * @brief FilterInvalidApplicationParameters filter app params that are not
+ * present in schema
+ * @param object of app policy params to filter
+ */
+void FilterInvalidApplicationParameters(
+    policy_table::ApplicationParams& app_params) {
+  // Filter AppHMIType array
+  policy_table::AppHMITypes valid_app_hmi_types;
+  policy_table::AppHMITypes app_hmi_types = *(app_params.AppHMIType);
+  for (const auto& app_hmi_type : app_hmi_types) {
+    if (app_hmi_type.is_valid()) {
+      valid_app_hmi_types.push_back(app_hmi_type);
+    }
+  }
+  app_params.AppHMIType->swap(valid_app_hmi_types);
+
+  // Filter RquestTypes array
+  policy_table::RequestTypes valid_request_types;
+  policy_table::RequestTypes request_types = *(app_params.RequestType);
+  for (const auto& request_type : request_types) {
+    if (request_type.is_valid()) {
+      valid_request_types.push_back(request_type);
+    }
+  }
+  if (valid_request_types.empty() && !request_types.empty()) {
+    // An empty RequestType array will allow all request types. No valid
+    // parameters are in the filtered array, so assign an uninitizlied value to
+    // for array to be "omitted"
+    policy_table::RequestTypes omitted_request_types;
+    *(app_params.RequestType) = omitted_request_types;
```

Comment:
@JackLivio make the same changes as for policy_external according to my comments there